src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -141,108 +82,155 @@ private static string GetReceiverTypeName(ITypeSymbol typeSymbol)
+                }
 
-            return (items, counter);
+                foreach (var peReference in GetAllRelevantPeReferences(currentProject))
+                {
+                    _ = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(solution, peReference, loadOnly: false, cancellationToken).ConfigureAwait(false);
+                }
+            }
         }
 
-        private static async Task<ImmutableArray<SerializableImportCompletionItem>> GetExtensionMethodItemsAsync(
+        private static IEnumerable<Project> GetAllRelevantProjects(Project project)
+        {
+            var solution = project.Solution;
+            var graph = solution.GetProjectDependencyGraph();
+            var relevantProjectIds = graph.GetProjectsThatThisProjectTransitivelyDependsOn(project.Id).Concat(project.Id);
+            return relevantProjectIds.Select(id => solution.GetRequiredProject(id)).Where(p => p.SupportsCompilation);
+        }
+
+        private static IEnumerable<PortableExecutableReference> GetAllRelevantPeReferences(Project project)
+            => project.MetadataReferences.OfType<PortableExecutableReference>();
+
+        private static async Task<ImmutableArray<IMethodSymbol>?[]> GetExtensionMethodSymbolsAsync(
             Document document,
-            ITypeSymbol receiverTypeSymbol,
-            ImmutableArray<string> receiverTypeNames,
-            GetIndicesResult indices,
             int position,
-            ISet<string> namespaceFilter,
+            ITypeSymbol receiverTypeSymbol,
+            ISet<string> namespaceInScope,
+            bool forceIndexCreation,
             StatisticCounter counter,
             CancellationToken cancellationToken)
         {
-            if (!indices.HasResult)
-            {
-                return ImmutableArray<SerializableImportCompletionItem>.Empty;
-            }
-
             var currentProject = document.Project;
+            var solution = currentProject.Solution;
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var currentCompilation = semanticModel.Compilation;
             var currentAssembly = currentCompilation.Assembly;
 
-            using var _1 = ArrayBuilder<IMethodSymbol>.GetInstance(out var methodsBuilder);
-            using var _2 = PooledDictionary<INamespaceSymbol, string>.GetInstance(out var namespaceNameCache);
-
             // This dictionary is used as cache among all projects and PE references. 
             // The key is the receiver type as in the extension method declaration (symbol retrived from current compilation).
             // The value indicates if we can reduce an extension method with this receiver type given receiver type.
             var checkedReceiverTypes = new ConcurrentDictionary<ITypeSymbol, bool>();
-
+            var receiverTypeNames = GetReceiverTypeNames(receiverTypeSymbol);
             receiverTypeNames = AttachComplexTypes(receiverTypeNames);
```

Comment:
Attach is a bit of an odd name.  Is there a reason for it instead of 'Add'?

Reply:
Fixed